include/nbl/builtin/hlsl/subgroup/fft.hlsl

```diff
@@ -0,0 +1,57 @@
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP_FFT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP_FFT_INCLUDED_
+
+#include "nbl/builtin/hlsl/fft/common.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/subgroup_basic.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/subgroup_shuffle.hlsl"
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace subgroup
+{
+    // TODO: specialize subgorupShuffleXor to different Scalar types. As of right now this would only support 1-wide scalars (no vectors)
+    template<typename Scalar, bool inverse>
+    void FFT_loop(uint32_t stride, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) {
+        const vector <Scalar, 4> loHiPacked = {lo.real(), lo.imag(), hi.real(), hi.imag()};
+        vector <Scalar, 4> shuffledLoHiPacked = glsl::subgroupShuffleXor< vector <Scalar, 4> > (loHiPacked, stride);
+        lo.real(shuffledLoHiPacked.x);
+        lo.imag(shuffledLoHiPacked.y);
+        hi.real(shuffledLoHiPacked.z);
+        hi.imag(shuffledLoHiPacked.w);
+        // Get twiddle with k = subgroupID mod stride, N = 2 * stride
+        fft::DIX<Scalar, inverse>::radix2(fft::twiddle<Scalar, inverse>(glsl::gl_SubgroupID() & (stride - 1), stride << 1), lo, hi);   
+    }
+
+
+    template<typename Scalar, uint32_t SubgroupSize, bool inverse>
+    void FFT(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) {
+        const uint32_t doubleSubgroupSize = SubgroupSize << 1;
+        // special first iteration
+        if (! inverse)
+            fft::DIX<Scalar, inverse>::radix2(fft::twiddle<Scalar, inverse>(glsl::gl_SubgroupID(), doubleSubgroupSize), lo, hi);                                                                                   
```

Comment:
using `DIX<Scalar,inverse>` when you're already inside an `if (inverse)` or `if (!inverse)` makes the code super unreadable, I'd use DIT or DIF directly